nominatim/tokenizer/sanitizers/delete_tags.py

```diff
@@ -0,0 +1,146 @@
+                 to be a candidate for removal if its 'kind' property
+                 fully matches any of the given regular expressions.
+
+    suffix: Define the 'suffix' property of the tags which should be
+            removed. Takes a string or list of strings where each
+            string is a regular expression. A tag is considered to be a
+            candidate for removal if its 'suffix' property fully
+            matches any of the given regular expressions.
+
+    name: Define the 'name' property corresponding to the 'kind' property
+          of the tag. Takes a string or list of strings where each string
+          is a regular expression. A tag is considered to be a candidate
+          for removal if its name fully matches any of the given regular
+          expressions.
+
+    country_code: Define the country code of places whose tags should be
+                  considered for removed. Takes a string or list of strings
+                  where each string is a two-letter lower-case country code.
+
+    rank_address: Define the address rank of places whose tags should be
+                  considered for removal. Takes a string or list of strings
+                  where each string is a number or range of number or the
+                  form <from>-<to>.
+                  Note that default is '0-30', which means that tags of all
+                  places are considered.
+                  See https://nominatim.org/release-docs/latest/customize/Ranking/#address-rank
+                  to learn more about address rank.
+
+
+"""
+from typing import Callable, List, Optional, Pattern, Tuple, Sequence
+import re
+
+from nominatim.tokenizer.sanitizers.base import ProcessInfo
+from nominatim.data.place_name import PlaceName
+from nominatim.tokenizer.sanitizers.config import SanitizerConfig
+
+class _TagSanitizer:
+
+    def __init__(self, config: SanitizerConfig) -> None:
+        self.type = config.get('type', 'name')
+        self.filter_kind = config.get_filter_kind()
+        self.country_codes = config.get_string_list('country_code', [])
+        self.rank_address_intervals = self._compute_rank_intervals( \
+                                            config.get_string_list('rank_address', ['0-30']))
+
+        suffixregexps = config.get_string_list('suffix', [])
+        self.suffix_regexp = [re.compile(r) for r in suffixregexps]
+
+        nameregexps = config.get_string_list('name', [])
+        self.name_regexp = [re.compile(r) for r in nameregexps]
+
+
+
+    def __call__(self, obj: ProcessInfo) -> None:
+        tags = obj.names if self.type == 'name' else obj.address
+
+        if not tags:
+            return
+
+        filtered_tags: List[PlaceName] = []
+
+        for tag in tags:
+            keep_tag = True
+
+            if self.filter_kind(tag.kind):
+
+                if (obj.place.country_code in self.country_codes and
+                    self._in_rank_addresses(obj.place.rank_address)):

```

Comment:
These two conditions apply to the whole object, not to the individual tags. That means, if they are not fulfilled, none of the tags will filtered. You can therefore have a shortcut and check those up in line 77 already.